clients/python-wrapper/lakefs/branch.py

```diff
@@ -356,11 +356,10 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         try:
-            if len(list(self._tx.uncommitted(amount=1))) == 0:
-                return  # Do nothing if tx is empty
-
-            self._tx_branch.commit(message=self._tx.commit_message, metadata=self._tx.commit_metadata)
-            self._tx_branch.merge_into(self._source_branch, message=f"Merge transaction {self._tx.id} to branch")
-            self._tx_branch.delete()
+            if len(list(self._tx.uncommitted(amount=1))) > 0:
+                self._tx_branch.commit(message=self._tx.commit_message, metadata=self._tx.commit_metadata)
+                self._tx_branch.merge_into(self._source_branch, message=f"Merge transaction {self._tx.id} to branch")
```

Comment:
This does a bit more than just clean up a branch, it ignores an error, is this expected? I would prefer if we have a special case for errors of type `no-changes` and remove the branch in that case.

Reply:
Though the code was logically correct, I moved it back into the try.
Can you elaborate on what errors are we ignoring? I don't think we need to treat no-changes as an error.
What is missing is dealing with exceptions raised while inside the context - which I added now. Thanks!